component/async/kafka/simple/simple.go

```diff
@@ -118,16 +118,14 @@ func (c *consumer) Consume(ctx context.Context) (<-chan async.Message, <-chan er
 					kafka.TopicPartitionOffsetDiffGaugeSet("", m.Topic, m.Partition, consumer.HighWaterMarkOffset(), m.Offset)
 					kafka.MessageStatusCountInc(kafka.MessageReceived, "", m.Topic)
 
-					go func(message *sarama.ConsumerMessage) {
-						msg, err := kafka.ClaimMessage(ctx, message, c.config.DecoderFunc, nil)
-						if err != nil {
-							kafka.MessageStatusCountInc(kafka.MessageClaimErrors, "", message.Topic)
-							chErr <- err
-							return
-						}
-						kafka.MessageStatusCountInc(kafka.MessageDecoded, "", message.Topic)
-						chMsg <- msg
-					}(m)
+					msg, err := kafka.ClaimMessage(ctx, m, c.config.DecoderFunc, nil)
+					if err != nil {
+						kafka.MessageStatusCountInc(kafka.MessageClaimErrors, "", m.Topic)
+						chErr <- err
+						return
```

Comment:
the more i think about it ...
it seems we could avoid return at this point

Reply:
Shall we? I'm puzzled... On one hand, I understand that it changes the current behaviour. On the other hand, the group consumer already stops the message consumption: https://github.com/beatlabs/patron/blob/726c0299dcf9f99b14357429e54b7eca89317c7c/component/async/kafka/group/group.go#L174

Reply:
yeah , it s not clear to me either what exactly this change will effect (probably also nothing).
Do you think we have some test the demonstrates this ?
Or should we add one ?